### PR TITLE
Replace Magma with Koe

### PR DESCRIPTION
**Pull request by freyacodes**
[Koe](https://github.com/KyokoBot/koe) is a smaller and perhaps more maintainable alternative to Magma.

Koe also relies on JDA-NAS natives for native audio sending.

### CODE REVIEW

**Comment by Dramex** [comment]
^ green circle lights up when joining a channel without playing a song.

**Comment by freyacodes** [comment]
@Dramex I haven't encountered that. Can you provide a log?

**Comment by Dramex** [comment]
@Frederikam https://pastebin.com/raw/WcpzmPMr

https://cdn.probot.io/IvVYoqIHry.png

~ updated with full log

**Comment by freyacodes** [comment]
@Dramex you're not playing anything.

Though it is odd that there is a green circle. Maybe that's because this version of Koe does not send silent frames.

**Comment by alula** [comment]
I'm working on it, Koe's handling of speaking is currently broken.

**Comment by alula** [comment]
@Frederikam https://github.com/KyokoBot/koe/releases/tag/1.0.0-beta1

**Comment by freyacodes** [comment]
Finally got around to fixing the breaking changes.

I haven't found the time to test it. Feedback is appreciated.

File: LavalinkServer/src/main/java/lavalink/server/player/Player.java [hunk 1]
```diff
@@ -185,4 +167,32 @@ public void onTrackStart(AudioPlayer player, AudioTrack track) {
         }
     }
 
+    public void provideTo(VoiceConnection connection) {
+        new Provider(connection);
```
Comment (Intuinewin) [inline LavalinkServer/src/main/java/lavalink/server/player/Player.java]:
```suggestion
        connection.setAudioSender(new Provider(connection));
```

**Review by Intuinewin — CHANGES_REQUESTED** [review]

File: LavalinkServer/src/main/java/lavalink/server/player/Player.java (hunk 1 above)
Comment (freyacodes) [reply LavalinkServer/src/main/java/lavalink/server/player/Player.java]: Nice catch

**Comment by freyacodes** [comment]
I am now using this in production. This is probably safe to use.

**Pull request merged**